### ISSUE TITLE
Is there any tips for download the QM9 Electron density functions? + Can you provide some preprocess codes for Cubic dataset? + etcs

### ISSUE THREAD

**Issue by seongsukim-ml** [ccr-cheng/InfGCN-pytorch#3]
Hello.

First of all, thank you for sharing your great work. I was impressed by your work and tried to reproduce your work to get insights.

But in the reproducing sequence, I had some troubles and wanted to ask about it.

1)

When I tried to download the QM9 EDF dataset in the figshares, it was about 1T and ETA was about 6 days in  my Linux systems.

In the downloading sequence, I used `wget` in the terminal but it was irregularly disconnected and started from the bottom.

Do you have any tips for downloading the QM9 EDF dataset?

2)

And I downloaded the Cubit datasets from Figshares but the number of the samples in the whole dataset(ECD-database-part1,2,3) was 17418.

However, in your paper, you used the 16421 dataset as shown in Appendix C.

Could you provide the list of the train/val/test split name or code to divide and read the dataset? I was stuck in the dataset split process.

3)

Finally, I got the reproduced results of ethane and malonaldehyde of the QM9. The results show that MAE as shown below:

Ethane with seed 42
![image](https://github.com/ccr-cheng/InfGCN-pytorch/assets/46878412/0eb7d01e-4f6b-466e-b738-070006d3d00a)

Malonaldehyde with seed 42
![image](https://github.com/ccr-cheng/InfGCN-pytorch/assets/46878412/ace4eab4-26e7-48c7-b27d-c8423e5010c1)

In the training metric, the final inference MAEs were about 0.12. Are these NMAE which was reported in the paper? I think it is quite different from the results in your main results.

So can you provide the seeds that you used in the paper or the pre-trained model for the MD? Or some detailed configuration of your model may be helpful, I am confused that I missed something.

Thank you.

---

```
# md_malon.yml
train:
  seed: 42 # default seed 42
  train_samples: 1024
  val_samples: 2048
  max_iter: 2000 # maximum iteration
  batch_size: 64
  log_freq: 10000
  val_freq: 20
  save_freq: 100
  max_grad_norm: 100.
  optimizer:
    type: adam
    lr: 5.e-3
    weight_decay: 0.
    beta1: 0.9
    beta2: 0.999
  scheduler:
    type: plateau
    factor: 0.5
    patience: 5
    min_lr: 1.e-5

test:
  batch_size: 16
  inf_samples: 4096
  num_infer: null
  num_vis: 2

datasets:
  type: small_density
  root: ../dataset
  mol_name: malonaldehyde

model:
  type: infgcn
  n_atom_type: 3
  num_radial: 16
  num_spherical: 7
  radial_embed_size: 64
  radial_hidden_size: 128
  num_radial_layer: 2
  num_gcn_layer: 3
  cutoff: 3.
  grid_cutoff: 3.
  is_fc: false
  gauss_start: 0.5
  gauss_end: 5.
```

**Comment by ccr-cheng** [comment]
Hi Seongsu! Thanks very much for your interest in our work. I'm willing to address your concerns as follows.

1. **Tips for downloading the dataset**. The bad news is that this dataset is indeed quite large. Here are some of my suggestions:
- Use multi-source downloaders like [aria2](https://aria2.github.io/) to accelerate the downloading.
- Use aria2 to support automatic resume if disconnected. `wget` seemed to get it wrong when I tried to resume downloading.
- Download individual tarfiles separately instead of a large 1T file. You may find the below file id useful:
```python
# 11 * 12 + 2 = 134 files
[
    31062047, 31062152, 31062214, 31062266, 31062311, 31062335, 31062359, 31062367, 31062371, 31062443, 31062520,
    31062584, 31062652, 31062736, 31062846, 31062937, 31063005, 31063081, 31063152, 31063213, 31063281, 31063321,
    31063402, 31063473, 31063534, 31063587, 31063660, 31063726, 31063846, 31063909, 31063996, 31064074, 31064161,
    31064251, 31064332, 31064416, 31064497, 31064584, 31064658, 31064743, 31064824, 31064889, 31064968, 31065040,
    31065109, 31065172, 31065244, 31065324, 31065388, 31065429, 31065445, 31065453, 31065457, 31065462, 31065466,
    31065468, 31065475, 31065495, 31065502, 31065518, 31065528, 31065533, 31065537, 31065545, 31065552, 31065554,
    31065558, 31065599, 31065608, 31065626, 31065641, 31065655, 31065710, 31065751, 31065764, 31065778, 31065791,
    31065811, 31065821, 31065835, 31065836, 31065850, 31065859, 31065860, 31065865, 31065881, 31065886, 31065890,
    31065892, 31065899, 31065907, 31065920, 31065931, 31065942, 31065949, 31065957, 31065970, 31065994, 31066008,
    31066012, 31066020, 31066030, 31066047, 31066054, 31066075, 31066084, 31066095, 31066108, 31066119, 31066120,
    31066128, 31066135, 31066149, 31066162, 31066176, 31066216, 31066224, 31066231, 31066236, 31066249, 31066263,
    31066303, 31066327, 31066344, 31066345, 31066347, 31066348, 31066365, 31066366, 31066371, 31066387, 31074558,
    31074568, 31074583,
]
```
Also, be sure to check your data integrity after downloading. Disconnection indeed happens often.

2. **Data split for Cubic**. We mentioned in Sec.5.1 that noble gases (He, Ne, Ar, Kr, Xe) were filtered out and we kept only the
crystal structures whose primitive cell contains less than 64 atoms. To facilitate your experiment, I provide the data split file for Cubic [here](https://uofi.box.com/s/8nfosxts1i8g643f8etdqqbtqlg5clhk). 

3. **Experimental setting for MD**. We apologize for the confusion but our model on the MD dataset was trained for 4k iterations instead. Also, we modified a minor bug regarding the parity when tidying up this code repo, which caused a slight difference in the final performance compared to the reported NMAE. In our rerunning of the experiments, our model still achieved state-of-the-art performance on the QM9 and Cubic datasets, but the numbers might have changed a bit.

I hope the above information can help you.


Best,
Chaoran

**Comment by seongsukim-ml** [comment]
Thank you very much for considering my questions.

I truly appreciate taking your time.

And above question 3, I got reasonable results when I used 4K epoch at the MD dataset.
